engine/src/main/java/org/terasology/rendering/backdrop/Skysphere.java

```diff
@@ -101,16 +101,30 @@ public float getSunPositionAngle() {
         return celSystem.getSunPosAngle();
     }
 
+    /**
+     * Get the strength of the light for the current day part, based on sun/moon visibility.
+     * @return the strength of the current light in [0;1] range.
+     */
     @Override
     public float getDaylight() {
-        float angle = (float) Math.toDegrees(TeraMath.clamp(Math.cos(getSunPositionAngle())));
-        float daylight = 1.0f;
+        float lightBasedOnTimeOfDay = 1.0f;
+        final float ambientLight = 0.2f; // not totally black when the sun goes down, but the moon is not yet up
+        double radSunAngle = getSunPositionAngle(); // expected [-PI;PI] for the day cycle to work correctly!
+
+        // if the degree of the sun/moon is less than "lessLightUnderAngle", then we gradually increase the light, instead of full intensity
+        final double lessLightUnderAngle = Math.sin(Math.toRadians(15.0));
+        double absSinSunAngle = Math.abs(Math.sin(radSunAngle));
+        if (absSinSunAngle < lessLightUnderAngle) {
+            lightBasedOnTimeOfDay = (float)(absSinSunAngle / lessLightUnderAngle);
+        }
+        // otherwise the light remains full intensity
 
-        if (angle < 24.0f) {
-            daylight = 1.0f - (24.0f - angle) / 24.0f;
+        // if the moon is out, we want to reduce the brightness
+        if(radSunAngle < 0.0) {
+            lightBasedOnTimeOfDay *= 0.2f;
         }
 
-        return daylight;
+        return ambientLight + (lightBasedOnTimeOfDay*(1.0f-ambientLight));
```

Comment:
In light of what I wrote above I'd rewrite the whole method as an if/elseif/elseif/else statement dependent on the main light angle. Also, I'd use linear interpolation rather than sin and cos to do the light intensity math. It would make sense to use trigonometry with a more realistic lighting model (i.e. taking in account atmospheric thickness) but in this case we just need to interpolate light intensities between 15/5 degrees and viceversa: I am 99% certain linear interpolation won't make a difference there.